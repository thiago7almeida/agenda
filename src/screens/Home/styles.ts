import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
  iconPlus: {height: 20, width: 20},
});
