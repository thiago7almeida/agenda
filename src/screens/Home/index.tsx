import React from 'react';
import {TopNavigation, Divider, Button, Icon} from '@ui-kitten/components';
import styles from './styles';
import ContentView from './components/ContentView';
import {SafeAreaView} from 'react-native-safe-area-context';
import {useNavigation} from '@react-navigation/native';

const IconPlus = () => {
  return <Icon name="plus-outline" fill="black" style={styles.iconPlus} />;
};

const ButtonRigth = () => {
  const navigation = useNavigation();
  return (
    <Button
      appearance="ghost"
      onPress={() => navigation.navigate('Details')}
      accessoryLeft={IconPlus}
    />
  );
};

const App: React.FC = () => {
  return (
    <SafeAreaView style={styles.container}>
      <TopNavigation
        title="Contatos"
        alignment={'center'}
        accessoryRight={ButtonRigth}
      />
      <Divider />
      <ContentView />
    </SafeAreaView>
  );
};

export default App;
