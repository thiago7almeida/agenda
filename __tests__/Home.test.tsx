import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Home from '../src/screens/Home';

describe('Home', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Primary', () => {
      const component = shallow(<Home />);
      expect(toJson(component)).toMatchSnapshot('Primary Home snapshot');
    });
  });
});
