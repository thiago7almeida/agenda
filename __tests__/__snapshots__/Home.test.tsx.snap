// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Rendering should match to snapshot - Primary: Primary Home snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <TopNavigation
    accessoryRight={[Function]}
    alignment="center"
    title="Contatos"
  />
  <Divider />
  <ContentView />
</RNCSafeAreaView>
`;
